Source/AwsCommonRuntimeKit/mqtt/Mqtt5Types.swift

```diff
@@ -712,35 +709,72 @@ public class NegotiatedSettings {
-        receiveMaximumFromServer: UInt16,
-        maximumPacketSizeToServer: UInt32,
-        topicAliasMaximumToServer: UInt16,
-        topicAliasMaximumToClient: UInt16,
-        serverKeepAlive: TimeInterval,
-        retainAvailable: Bool,
-        wildcardSubscriptionsAvailable: Bool,
-        subscriptionIdentifiersAvailable: Bool,
-        sharedSubscriptionsAvailable: Bool,
-        rejoinedSession: Bool,
-        clientId: String) {
-
-            self.maximumQos = maximumQos
-            self.sessionExpiryInterval = sessionExpiryInterval
-            self.receiveMaximumFromServer = receiveMaximumFromServer
-            self.maximumPacketSizeToServer = maximumPacketSizeToServer
-            self.topicAliasMaximumToServer = topicAliasMaximumToServer
-            self.topicAliasMaximumToClient = topicAliasMaximumToClient
-            self.serverKeepAlive = serverKeepAlive
-            self.retainAvailable = retainAvailable
-            self.wildcardSubscriptionsAvailable = wildcardSubscriptionsAvailable
-            self.subscriptionIdentifiersAvailable = subscriptionIdentifiersAvailable
-            self.sharedSubscriptionsAvailable = sharedSubscriptionsAvailable
-            self.rejoinedSession = rejoinedSession
-            self.clientId = clientId
+    public init (maximumQos: QoS,
+                 sessionExpiryInterval: TimeInterval,
+                 receiveMaximumFromServer: UInt16,
+                 maximumPacketSizeToServer: UInt32,
+                 topicAliasMaximumToServer: UInt16,
+                 topicAliasMaximumToClient: UInt16,
+                 serverKeepAlive: TimeInterval,
+                 retainAvailable: Bool,
+                 wildcardSubscriptionsAvailable: Bool,
+                 subscriptionIdentifiersAvailable: Bool,
+                 sharedSubscriptionsAvailable: Bool,
+                 rejoinedSession: Bool,
+                 clientId: String) {
+        self.maximumQos = maximumQos
+        self.sessionExpiryInterval = sessionExpiryInterval
+        self.receiveMaximumFromServer = receiveMaximumFromServer
+        self.maximumPacketSizeToServer = maximumPacketSizeToServer
+        self.topicAliasMaximumToServer = topicAliasMaximumToServer
+        self.topicAliasMaximumToClient = topicAliasMaximumToClient
+        self.serverKeepAlive = serverKeepAlive
+        self.retainAvailable = retainAvailable
+        self.wildcardSubscriptionsAvailable = wildcardSubscriptionsAvailable
+        self.subscriptionIdentifiersAvailable = subscriptionIdentifiersAvailable
+        self.sharedSubscriptionsAvailable = sharedSubscriptionsAvailable
+        self.rejoinedSession = rejoinedSession
+        self.clientId = clientId
+    }
+
+    static func convertFromNative(_ from: UnsafePointer<aws_mqtt5_negotiated_settings>?) -> NegotiatedSettings? {
+
+        if let from = from {
+            guard let negotiatedMaximumQos = QoS(rawValue: Int(from.pointee.maximum_qos.rawValue))
```

Comment:
Similar to above: we could improve by 
```
let view = from.pointee
```

Reply:
Waqar mentioned we could use a private init instead of static function `convertFromNative`. I think that is a good idea.